tests/integration/failPolicy_test.go

```diff
@@ -0,0 +1,79 @@
+package integration_test
+
+import (
+	"context"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/rancher/lasso/pkg/client"
+	v3 "github.com/rancher/rancher/pkg/apis/cluster.cattle.io/v3"
+	provisioningv1 "github.com/rancher/rancher/pkg/apis/provisioning.cattle.io/v1"
+	"github.com/rancher/wrangler/pkg/gvk"
+	"github.com/rancher/wrangler/pkg/kubeconfig"
+	"github.com/rancher/wrangler/pkg/schemes"
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/suite"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type FailurePolicySuite struct {
+	suite.Suite
+	clientFactory client.SharedClientFactory
+}
+
+// TestFailurePolicyTest should be run only when the webhook is not running.
+func TestFailurePolicyTest(t *testing.T) {
+	suite.Run(t, new(FailurePolicySuite))
+}
+
+func (m *FailurePolicySuite) SetupSuite() {
+	logrus.SetLevel(logrus.DebugLevel)
+	kubeconfigPath := os.Getenv("KUBECONFIG")
+	logrus.Infof("Setting up test with KUBECONFIG=%s", kubeconfigPath)
+	restCfg, err := kubeconfig.GetNonInteractiveClientConfig(kubeconfigPath).ClientConfig()
+	m.Require().NoError(err, "Failed to clientFactory config")
+	m.clientFactory, err = client.NewSharedClientFactoryForConfig(restCfg)
+	m.Require().NoError(err, "Failed to create clientFactory Interface")
+
+	schemes.Register(v3.AddToScheme)
+	schemes.Register(provisioningv1.AddToScheme)
+	schemes.Register(corev1.AddToScheme)
+
+}
+
+func (m *FailurePolicySuite) TestNamespaceFail() {
+	const testNamespace = "test-namespace"
+	validCreateObj := &corev1.Namespace{
+		ObjectMeta: v1.ObjectMeta{
+			Name: testNamespace,
+		},
+	}
+
+	objGVK, err := gvk.Get(validCreateObj)
+	m.Require().NoError(err, "failed to get GVK")
+
+	client, err := m.clientFactory.ForKind(objGVK)
+	m.Require().NoError(err, "Failed to create client")
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*2)
+	defer cancel()
+	err = client.Create(ctx, "", validCreateObj, nil, v1.CreateOptions{})
+	m.Require().True(errors.IsInternalError(err), "Webhook should fail with service unavailable when the webhook is down instead we received :%v", err)
+
+	// attempt to clean up namespace if the create went through
+	defer client.Delete(ctx, "", testNamespace, v1.DeleteOptions{})
+
+	err = client.Update(ctx, "", validCreateObj, nil, v1.UpdateOptions{})
+	m.Require().True(errors.IsInternalError(err) || errors.IsNotFound(err), "Webhook should fail with service unavailable when the webhook is down instead we received :%v", err)
```

Comment:
Nit: I don't think that `NotFound` would be a valid condition for this. That would indicate to me that the webhook did not reject the request, and instead this went through. Meaning that the fail condition was `ignore`, when we want it to be `fail` on update calls.